src/model/sample.rs

```diff
@@ -36,40 +36,73 @@ pub fn prob_read_base(read_base: u8, ref_base: u8, base_qual: u8) -> LogProb {
     }
 }
 
-
 /// For an SNV, calculate likelihood of ref (first) or alt (second) for a given read, i.e. Pr(Z_i | ref) and Pr(Z_i | alt).
 /// This follows Samtools and GATK.
-pub fn prob_read_snv(record: &bam::Record, cigar: &[Cigar], start: u32, variant: Variant, ref_seq: &[u8]) -> (LogProb, LogProb) {
-    let contains_start = |pos: i32, cigar_length: u32|  pos as u32 <= start && pos as u32 + cigar_length >= start;
+/// TODO: Add cigar string parsing tests for SNVs.
+/// TODO: Implement more general cigar parsing function analogous to htslib resolve_cigar2 (sam.c) somewhere more central in libprosic or rust-htslib.
+pub fn prob_read_snv(record: &bam::Record, cigar: &[Cigar], vpos: u32, variant: Variant, ref_seq: &[u8]) -> Option<(LogProb, LogProb)> {
+    let contains_vpos = |pos: u32, cigar_length: u32|  pos <= vpos && pos + cigar_length > vpos;
 
     if let Variant::SNV(base) = variant {
-        let mut pos = record.pos();
-        let mut qpos = 0;
-        for c in cigar {
-            match c {
+        let mut rpos = record.pos() as u32; // reference position
+        let mut qpos = 0u32; // position within query / read / observation
+        let mut c = 0; // index into cigar operation vector
+        // find first cigar operation referring to qpos = 0 (and thus bases in record.seq()), because
+        // all augmentations of qpos and rpos before that are invalid
+        for i in 0..cigar.len() {
+            match &cigar[i] {
+                &Cigar::Match(_) |
+                &Cigar::Diff(_)  |
+                &Cigar::Equal(_) |
+                &Cigar::Ins(_)   | // this is unexpected, but bwa + GATK indel realignment can produce insertions before matching positions
+                &Cigar::SoftClip(_) => {
+                    c = i;
+                    break;
+                },
+                &Cigar::Del(_) => panic!("Cigar String: Unexpected Deletion operation found before any reference match."),
```

Comment:
For now let's leave it in here. We can always do that later, but since there is currently no functionality within Rust-Htslib to use it, I don't think it would be the right place for now.